source/scripts/app.js

```diff
@@ -245,12 +303,41 @@ function bindExploreLoadButton() {
   let topLevel = shadow.getElementById("explore-top-level");
   let loadButton = shadow.getElementById("load-button");
 
+  let vegan = shadow.getElementById("vegan");
+  let glutenFree = shadow.getElementById("gluten-free");
+  let vegetarian = shadow.getElementById("vegetarian");
+  let input = shadow.getElementById("search-bar");
+
   loadButton.addEventListener("click", async () => {
-    if (topLevel.classList.contains("type-explore")) {
+    if (topLevel.classList.contains("type-explore") && input.value === "") {
```

Comment:
Should we also require that the other filters be empty? Because if we are in `type-explore` and we have some filters checked, then clicking the `load-button` should return recipes that match those filters, not just random recipes that don't match the filters.
```suggestion
    if (topLevel.classList.contains("type-explore") &&
      input.value === "" &&
      !vegan.checked &&
      !glutenFree.checked &&
      !vegetarian.checked
    ) {
```

Reply:
Ah, yes, I forgot to add that here. Thanks!